src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1166,18 +1185,20 @@ public boolean checkOrRequestItems(@Nullable final ItemStack... items)
      * Please stop execution on false returned.
      *
      * @param useItemDamage compare the itemDamage of the values.
+     * @param waitForRequest wait for the request.
      * @param items         the items needed
      * @return false if they are in inventory
      */
-    public boolean checkOrRequestItems(final boolean useItemDamage, @Nullable final ItemStack... items)
+    public boolean checkOrRequestItems(final boolean useItemDamage, boolean waitForRequest, @Nullable final ItemStack... items)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "useItemDamage" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)